### PR TITLE
fix(glimmer-wrapper): handle addon local lookup paths

### LINKED ISSUE #296: Local Lookup does not work for addons
Originally reported as https://github.com/emberjs/ember.js/issues/17395.

This code does not account for addons / paths that already have a `rootName`:

https://github.com/ember-cli/ember-resolver/blob/df3ef4e3b327b4cb70badafbe973b4c8c2aa7051/mu-trees/addon/resolvers/glimmer-wrapper/index.js#L96-L109

Specifically, L106 does not anticipate a `source` that looks like:

```
template:addon-local-lookup/src/ui/components/foo/template.hbs
```

### PR DESCRIPTION
**Pull request by buschtoens**
Fixes #296.

### CODE REVIEW

File: mu-trees/addon/resolvers/glimmer-wrapper/index.js [hunk 1]
```diff
@@ -103,9 +103,10 @@ const Resolver = GlobalsResolver.extend({
         source = `${type}:/${rootName}/`;
       } else if (source) {
         // make absolute
-        let parts = source.split(':src/ui/');
-        source = `${parts[0]}:/${rootName}/${parts[1]}`;
-        source = source.split('/template.hbs')[0];
+        // For apps:              template:src/ui/components/foo/template.hbs
+        // For addons: template:addon-name/src/ui/components/foo/template.hbs
+        let [, type, root = rootName, name] = /^([^:]+):(?:([^/]+)\/)?src\/ui\/(.+)(?:\/template\.hbs)$/.exec(source);
```
Comment (buschtoens) [inline mu-trees/addon/resolvers/glimmer-wrapper/index.js]: Forgot to handle scoped packages.

```suggestion
        let [, type, root = rootName, name] = /^([^:]+):(?:((?:@[^/]+/)?[^/]+)\/)?src\/ui\/(.+)(?:\/template\.hbs)$/.exec(source);
```

**Comment by buschtoens** [comment]
In order for scoped packages to work correctly, the consuming project needs to have https://github.com/glimmerjs/glimmer-di/pull/46, which has been released as `@glimmer/di@0.2.1`. Since `@glimmer/di` is just a transitive dependency of [`@glimmer/resolver`](https://github.com/glimmerjs/glimmer-resolver/blob/master/package.json#L26), which is still specifying `^0.2.0`, you might need to force a resolution.

**Comment by buschtoens** [comment]
https://github.com/ember-cli/ember-resolver/pull/270#pullrequestreview-178911802 probably applies here as well.

**Comment by buschtoens** [comment]
I don't think there's really any way to avoid using a regular expression here. The only possible perf optimization I can think of is keeping the `split` logic for the app / non-addon code path and only falling back to the `ADDON_SPECIFIER` regex, if `split` does not return an array of 2 elements.

@rwjblue What do you think?

**Comment by buschtoens** [comment]
I've submitted #340 which implements what I described above.

**Comment by NullVoxPopuli** [comment]
this fixed local lookup issues emberclear was having for the past month or so. :)

**Comment by stefanpenner** [comment]
As module unification is no more, I believe this can be closed. If I am closing this in error please let me know why, and we can reopen.

**Pull request closed without merging**